keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/SecurityGroupRuleComparator.kt

```diff
@@ -114,66 +114,7 @@ private fun SecurityGroup.canonicalize(): SecurityGroup {
     .newBuilder(this).apply {
       clearInboundRule()
       originalInboundRuleList.sortedWith(SecurityGroupRuleComparator).forEach { originalRule ->
-        addInboundRule(originalRule.toBuilder().apply {
-          when {
-            hasReferenceRule() ->
-              referenceRule = referenceRule.canonicalize()
-            hasCrossRegionReferenceRule() ->
-              crossRegionReferenceRule = crossRegionReferenceRule.canonicalize()
-            hasSelfReferencingRule() ->
-              selfReferencingRule = selfReferencingRule.canonicalize()
-            hasCidrRule() ->
-              cidrRule = cidrRule.canonicalize()
-          }
-        }.build())
-      }
-    }
-    .build()
-}
-
-private fun ReferenceRule.canonicalize(): ReferenceRule {
-  val originalPortRangeList = portRangeList
-  return toBuilder()
-    .apply {
-      clearPortRange()
-      originalPortRangeList.sortedWith(PortRangeComparator).forEach {
-        addPortRange(it)
-      }
-    }
-    .build()
-}
-
-private fun CrossRegionReferenceRule.canonicalize(): CrossRegionReferenceRule {
-  val originalPortRangeList = portRangeList
-  return toBuilder()
-    .apply {
-      clearPortRange()
-      originalPortRangeList.sortedWith(PortRangeComparator).forEach {
-        addPortRange(it)
-      }
-    }
-    .build()
-}
-
-private fun SelfReferencingRule.canonicalize(): SelfReferencingRule {
-  val originalPortRangeList = portRangeList
-  return toBuilder()
-    .apply {
-      clearPortRange()
-      originalPortRangeList.sortedWith(PortRangeComparator).forEach {
-        addPortRange(it)
-      }
-    }
-    .build()
-}
-
-private fun CidrRule.canonicalize(): CidrRule {
-  val originalPortRangeList = portRangeList
-  return toBuilder()
-    .apply {
-      clearPortRange()
-      originalPortRangeList.sortedWith(PortRangeComparator).forEach {
-        addPortRange(it)
+        addInboundRule(originalRule)
```

Comment:
Canonicalization is much easier now as I just have to order the rules not the rules _and_ the port ranges inside them.